internal/image/image.go

```diff
@@ -5,25 +5,38 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 
 	"github.com/docker/app/internal/slices"
 	composetypes "github.com/docker/cli/cli/compose/types"
+	"github.com/docker/distribution/reference"
 	"github.com/pkg/errors"
 )
 
 // Add add service images to the app package
-func Add(appname string, services []string, config *composetypes.Config) error {
-	if err := os.Mkdir(filepath.Join(appname, "images"), 0755); err != nil {
-		return errors.Wrap(err, "cannot create 'images' folder")
+func Add(appname string, services []string, config *composetypes.Config, pull bool, quiet bool) error {
+	if err := os.Mkdir(filepath.Join(appname, "images"), 0755); err != nil && !os.IsExist(err) {
+		return errors.Wrap(err, "cannot create 'images' folder, this command only works with a multi-file application package")
 	}
 	for _, s := range config.Services {
 		if len(services) != 0 && !slices.ContainsString(services, s.Name) {
 			continue
 		}
+		if !quiet {
+			fmt.Printf("Adding image %s for service %s...\n", s.Image, s.Name)
+		}
+		if pull {
+			cmd := exec.Command("docker", "pull", s.Image)
```

Comment:
Using `docker pull` means that we won't be able to fetch Windows images on Linux machines. This is OK for now but we'll need a generic solution before it's pulled out of experimental.

cc @garethr

Reply:
I'd probably look to do that now if we can. I worry about things like that becoming hidden and then either falling through or causing a delay because we forgot.